validation/proxy.go

```diff
@@ -0,0 +1,243 @@
+package validation
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"sync"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/openapi3filter"
+	"github.com/getkin/kin-openapi/routers"
+	"github.com/getkin/kin-openapi/routers/gorillamux"
+	"github.com/mitchellh/copystructure"
+
+	"github.com/kubeshop/kusk-gateway/options"
+)
+
+var (
+	reAdjustSubstitutions = regexp.MustCompile(`\\(\d+)`)
+)
+
+const (
+	HeaderServiceID   = "X-Kusk-Service-ID"
+	HeaderOperationID = "X-Kusk-Operation-ID"
+)
```

Comment:
We can unexport these 2 constants if they are only used within this package

Reply:
I think we should be moving a lot of our packages into "internal" directory unless we specifically want it to be exported for other packages. It becomes overhelming to understand what is where for the new reader.

Reply:
Resolving this because they are actually used in Envoy config generator now (they were hard-coded there before).